ludwig/datasets/mixins/gzip_download.py

```diff
@@ -0,0 +1,68 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import gzip
+import os
+import requests
+
+
+class GZipDownloadMixin:
+    """Downloads a gzip file containing the mnist png dataset and extracts its contents."""
+
+    config: dict
+    raw_dataset_path: str
+    raw_temp_path: str
+
+    def download_raw_dataset(self):
```

Comment:
Since this logic is specific to the MNIST dataset, it should be moved into the `Mnist` class.  Mixins should be used only for generic functionality applicable to multiple datasets.